server/lib/nl/nl_utils.py

```diff
@@ -89,4 +92,107 @@ def combine_stop_words() -> Set[str]:
 def remove_punctuations(s):
   s = s.replace('\'s', '')
   s = re.sub(r'[^\w\s]', ' ', s)
-  return " ".join(s.split())
\ No newline at end of file
+  return " ".join(s.split())
+
+
+def is_topic(sv):
+  return sv.startswith("dc/topic/")
+
+
+def is_svg(sv):
+  return sv.startswith("dc/g/")
+
+
+def is_sv(sv):
+  return not (is_topic(sv) or is_svg(sv))
+
+
+#
+# Returns a list of existing SVs (as a union across places).
+#
+def sv_existence_for_places(places: List[str], svs: List[str]) -> List[str]:
+  if not svs:
+    return []
+
+  sv_existence = dc.observation_existence(svs, places)
+  if not sv_existence:
+    logging.error("Existence checks for SVs failed.")
+    return []
+
+  existing_svs = set()
+  for sv in svs:
+    for _, exist in sv_existence['variable'][sv]['entity'].items():
+      if not exist:
+        continue
+      existing_svs.add(sv)
+
+  return list(existing_svs)
+
+
+_NUM_CHILD_PLACES_FOR_EXISTENCE = 20
+
+
+#
+# Given a place DCID and a child place type, returns a sample list
+# of places of that child type.
+#
+# TODO: Maybe dedupe with nl_data_spec.py
+#
+def get_sample_child_places(main_place_dcid: str,
+                            contained_place_type: str) -> List[str]:
+  """Find a sampled child place"""
+  logging.info('_sample_child_place: for %s - %s', main_place_dcid,
+               contained_place_type)
+  if not contained_place_type:
+    return None
+  if contained_place_type == "City":
+    return "geoId/0667000"
```

Comment:
maybe we should pick a "simpler" city (one that's not also a county)

Reply:
SG. picked sunnyvale (not a county, nor a common name)